pkg/ccl/backupccl/tenant_backup_nemesis_test.go

```diff
@@ -0,0 +1,505 @@
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "IMPORT INTO",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("import_into")
+					if err != nil {
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "CREATE INDEX",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("create_index")
+					if err != nil {
+						return err
+					}
+					if _, err := db.Exec(fmt.Sprintf(`CREATE INDEX ON "%s"(n)`, tableName)); err != nil {
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "CREATE UNIQUE INDEX (will fail)",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("create_unique_index")
+					if err != nil {
+						return err
+					}
+					// Add a conflict that ensures the unique index creation will fail.
+					mkConflict := fmt.Sprintf(`UPDATE "%[1]s" SET n = 5 WHERE id IN (SELECT id FROM "%[1]s" ORDER BY random() LIMIT 5)`, tableName)
+					if _, err := db.Exec(mkConflict); err != nil {
+						return err
+					}
+
+					if _, err := db.Exec(fmt.Sprintf(`CREATE UNIQUE INDEX ON "%s"(n)`, tableName)); err == nil {
+						return errors.New("expected error but found none")
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+		},
+	}
+}
+
+func (r *randomBackupNemesis) Start(ctx context.Context) {
+	ctx, cancel := context.WithCancel(ctx)
+
+	r.cancel = cancel
+	r.grp = ctxgroup.WithContext(ctx)
+
+	r.grp.GoCtx(r.runNemesis)
+}
+
+func (r *randomBackupNemesis) Stop() {
+	r.cancel()
+	_ = r.grp.Wait()
+}
+
+func (r *randomBackupNemesis) RequireStart() chan struct{} {
+	notifyCh := make(chan nemesisNotification)
+	r.registerOneTimeListener(notifyCh)
+	newNemesis := <-notifyCh
+	return newNemesis.done
+}
+
+func (r *randomBackupNemesis) TablesToCheck() []string {
+	r.mu.Lock()
+	ret := append([]string(nil), r.mu.tablesToCheck...)
+	r.mu.Unlock()
+	return ret
+}
+
+func (r *randomBackupNemesis) addTable(name string) {
+	r.mu.Lock()
+	r.mu.tablesToCheck = append(r.mu.tablesToCheck, name)
+	r.mu.Unlock()
+}
+
+func (r *randomBackupNemesis) registerOneTimeListener(ch chan nemesisNotification) {
+	r.mu.Lock()
+	r.mu.oneTimeListeners = append(r.mu.oneTimeListeners, ch)
+	r.mu.Unlock()
+}
+
+func (r *randomBackupNemesis) notifyOneTimeListeners(done chan struct{}) {
+	r.mu.Lock()
+	for _, l := range r.mu.oneTimeListeners {
+		l <- nemesisNotification{
+			done: done,
+		}
+	}
+	r.mu.oneTimeListeners = nil
+	r.mu.Unlock()
+}
+
+func (r *randomBackupNemesis) runNemesis(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+		n := r.nemeses[r.rng.Intn(len(r.nemeses))]
+		doneCh := make(chan struct{})
+		r.notifyOneTimeListeners(doneCh)
```

Comment:
what's the point of this `r.NotifyOneTimeListerners` call?

Reply:
it seems like the only thing that could be listening is the listener registered during RequireStart(), though its hard for me to understand what interaction this notification call has with that listener.

Reply:
notifyOneTimeListeners unblocks anyone who was waiting for RequiredStart().